Automattic/wp-calypso
# Plans: Implement formatCurrency for header-billing-info section

This PR implements formatCurrency in the 2023 plans grid "header-billing-info" section:

![image](https://user-images.githubusercontent.com/16580129/226785754-eb5c21b6-6c87-4cbe-842e-a03a20998efc.png)

## Testing Instructions

* Set your locale from `/me` to `fr-CA`
* Go to `/plans/[your-site-url]`
* You should now see the currency symbol right aligned

## Pre-merge Checklist


Related to #74605
